src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordConstructor.cs

```diff
@@ -36,13 +33,11 @@ internal RecordDeclarationSyntax GetSyntax()
             return GetSyntax().PrimaryConstructorBaseType;
         }
 
-        internal override bool IsExpressionBodied
-        {
-            get
-            {
-                return false;
-            }
-        }
+        protected override bool AllowRefOrOut => false;
+
+        internal override bool IsExpressionBodied => false;
+
+        public override bool IsImplicitlyDeclared => true;
```

Comment:
>public override bool IsImplicitlyDeclared => true; [](start = 8, length = 50)

Please revert this change. This method is not implicitly declared. We had discussion about this in contex of the PR that changed base type for this class.

Reply:
Also, I believe this change is way outside of the advertised goal of this PR. 

---
In reply to: [454451705](https://github.com/dotnet/roslyn/pull/45898#discussion_r454451705) [](ancestors = 454451705)

Reply:
That seems wrong to me, but if it's already been discussed and resolved then it's okay. Where did the discussion happen? 
I didn't see it in [https://github.com/dotnet/roslyn/pull/44803](https://github.com/dotnet/roslyn/pull/44803)

---
In reply to: [454465241](https://github.com/dotnet/roslyn/pull/45898#discussion_r454465241) [](ancestors = 454465241,454451705)

Reply:
>That seems wrong to me, but if it's already been discussed and resolved then it's okay. Where did the discussion happen? 
I didn't see it in [https://github.com/dotnet/roslyn/pull/44803](https://github.com/dotnet/roslyn/pull/44803)

You are correct, it is quite possible I confused it with this thread [https://github.com/dotnet/roslyn/pull/44803#pullrequestreview-423731255](https://github.com/dotnet/roslyn/pull/44803#pullrequestreview-423731255). In any case, there is a syntax explicitly declaring the constructor - at least the parameter list. There is also other user code in it - base arguments. Likely there will be a body in the future as well. Even today we should fire analyzers actions for the constructor, due to presence of user code. Analyzer driver filters out implicitly declared symbols all over the code.

---
In reply to: [454547147](https://github.com/dotnet/roslyn/pull/45898#discussion_r454547147) [](ancestors = 454547147,454465241,454451705)